test/system/252-quadlet.bats

```diff
@@ -1599,4 +1603,58 @@ EOF
     run_podman rmi $untagged_image:latest $built_image $(pause_image)
     run_podman network rm podman-default-kube-network
 }
+
+@test "quadlet - drop-in files" {
+    local quadlet_tmpdir="${PODMAN_TMPDIR}/dropins"
+
+    local quadlet_file="truncated-$(random_string).container"
+
+    local -A dropin_dirs=(
+        [toplevel]=container.d
+        [truncated]=truncated-.container.d
+        [quadlet]="${quadlet_file}.d"
+    )
+
+    # Table of drop-in .conf files. Format is:
+    #
+    #    apply | dir | filename | [Section] | Content=...
+    local dropin_files="
+y | toplevel  | 10 | [Unit]      | Description=Test File for Dropin Configuration
+n | toplevel  | 99 | [Install]   | WantedBy=default.target
+y | truncated | 50 | [Container] | ContainerName=truncated-dropins
+n | truncated | 99 | [Service]   | Restart=always
+n | truncated | 99 | [Install]   | WantedBy=multiuser.target
+y | quadlet   | 99 | [Service]   | RestartSec=60s
+"
+
+    # Pass 1: Create all drop-in directories and files
+    while read apply dir file section content; do
+        local d="${quadlet_tmpdir}/${dropin_dirs[${dir}]}"
+        mkdir -p "${d}"
+
+        local f="${d}/${file}.conf"
+        echo "${section}" >>"${f}"
+        echo "${content}" >>"${f}"
+    done < <(parse_table "${dropin_files}")
+
+    # Create the base quadlet file
+    quadlet_base="${PODMAN_TMPDIR}/${quadlet_file}"
+    cat > "${quadlet_base}" <<EOF
+[Container]
+Image="${IMAGE}"
+EOF
+
+    # Generate the quadlet file from the base file and any drop-in .conf files.
+    run_quadlet "${quadlet_base}" "${quadlet_tmpdir}"
+
+    # Pass 2: test whether the expected .conf files are applied
+    # and the overridden .conf files are not.
+    while read apply dir file section content; do
+        if [[ "${apply}" = "y" ]]; then
+            assert "${QUADLET_SERVICE_CONTENT}" =~ "${content}" "Set in ${dir}/${file}.conf"
+        else
+            assert "${QUADLET_SERVICE_CONTENT}" !~ "${content}" "Set in ${dir}/${file}.conf but should have been overridden"
+        fi
+    done < <(parse_table "${dropin_files}")
```

Comment:
It was intentional -- I didn't think it was necessary since we iterate through the whole table, so `WantedBy=default.target` and `WantedBy=multiuser.target` are already checked, and since neither the base file nor any of the drop-in files adds a `WantedBy=`, there was no need to check again.

I had it in originally since it was a regex, so it handled both instances of when the `WantedBy=` may get added.

Just for clarity, both get overridden because only the lowest drop-in .conf in the hierarchy gets applied. Since `y | quadlet   | 99 | [Service]   | RestartSec=60s` is defined, none of `n | toplevel  | 99 | [Install]   | WantedBy=default.target`,  `n | truncated | 99 | [Service]   | Restart=always`, and
`n | truncated | 99 | [Install]   | WantedBy=multiuser.target` are applied since they are in equivalently named files higher up the hierarchy. In the code, they aren't ever even parsed.